File: GENERECIDAD/EJERCICIO-1/Caja.py
from typing import Generic, TypeVar

T = TypeVar('T')

class Caja(Generic[T]):
    def __init__(self):
        self.cont: T = None
    def guardar(self,cont: T):
        self.cont = cont
    def obtener(self)-> T:
        return self.cont

caja_txt = Caja[str]()
caja_txt.guardar("holaaaa")
caja_numero = Caja[int]()
caja_numero.guardar(2567)

print("Contenido de caja texto: ", caja_txt.obtener())
print("contenido de caja numero", caja_numero.obtener())

File: GENERECIDAD/EJERCICIO-1/Caja.java
//package Generecidad_1;

class Caja <T> {
	private T cont;
	public void guardar(T cont) {
		this.cont = cont;
	}
	public T obtener() {
		return cont;
	}
	public static void main(String[] args) {
		Caja<String> cajaDeTexto = new Caja<>();
		cajaDeTexto.guardar("holaa");
		
		Caja<Integer> cajaDeEntero = new Caja<>();
		cajaDeEntero.guardar(1243);
		System.out.println("contentide de caja texto "+ cajaDeTexto.obtener());
		System.out.println("Contednido de caja entero "+ cajaDeTexto.obtener());
	}
}
